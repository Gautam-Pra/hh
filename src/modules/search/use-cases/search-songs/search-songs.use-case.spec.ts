import { beforeAll, describe, expect, it } from 'vitest'
import { SearchSongsUseCase } from '#modules/search/use-cases'

describe('SearchSongs', () => {
  let searchSongsUseCase: SearchSongsUseCase

  beforeAll(() => {
    searchSongsUseCase = new SearchSongsUseCase()
  })

  it('should search songs by query and return a list of songs', async () => {
    const songs = await searchSongsUseCase.execute({ query: 'ride', limit: 5, page: 1 })

    expect(songs).toMatchSnapshot({
      total: expect.any(Number),
      results: songs.results.map((song) => ({
        ...song,
        playCount: expect.any(Number)
      }))
    })
  })
})
