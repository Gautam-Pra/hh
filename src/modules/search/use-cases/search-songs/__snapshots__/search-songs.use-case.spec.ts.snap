// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchSongs > should search songs by query and return a list of songs 1`] = `
{
  "results": [
    {
      "album": {
        "id": "1712199",
        "name": "My Own Way",
        "url": "https://www.jiosaavn.com/album/my-own-way/on14RdH8SyM_",
      },
      "artists": {
        "all": [
          {
            "id": "464972",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_500x500.jpg",
              },
            ],
            "name": "Jay Sean",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/jay-sean-songs/8Ea5vNMFq7w_",
          },
        ],
        "featured": [],
        "primary": [
          {
            "id": "464972",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_500x500.jpg",
              },
            ],
            "name": "Jay Sean",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/jay-sean-songs/8Ea5vNMFq7w_",
          },
        ],
      },
      "copyright": "(C) 2008 2Point9/Jayded",
      "downloadUrl": [
        {
          "quality": "12kbps",
          "url": "https://aac.saavncdn.com/048/5b0e6da1abf670f7205f701d67374b8f_12.mp4",
        },
        {
          "quality": "48kbps",
          "url": "https://aac.saavncdn.com/048/5b0e6da1abf670f7205f701d67374b8f_48.mp4",
        },
        {
          "quality": "96kbps",
          "url": "https://aac.saavncdn.com/048/5b0e6da1abf670f7205f701d67374b8f_96.mp4",
        },
        {
          "quality": "160kbps",
          "url": "https://aac.saavncdn.com/048/5b0e6da1abf670f7205f701d67374b8f_160.mp4",
        },
        {
          "quality": "320kbps",
          "url": "https://aac.saavncdn.com/048/5b0e6da1abf670f7205f701d67374b8f_320.mp4",
        },
      ],
      "duration": 190,
      "explicitContent": false,
      "hasLyrics": false,
      "id": "IRytJK-g",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/048/My-Own-Way-English-2015-50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/048/My-Own-Way-English-2015-150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/048/My-Own-Way-English-2015-500x500.jpg",
        },
      ],
      "label": "2point9",
      "language": "english",
      "lyricsId": null,
      "name": "Ride It",
      "playCount": Any<Number>,
      "releaseDate": null,
      "type": "song",
      "url": "https://www.jiosaavn.com/song/ride-it/OToSRT57GlQ",
      "year": 2015,
    },
    {
      "album": {
        "id": "17679595",
        "name": "My Own Way (Hindi Version)",
        "url": "https://www.jiosaavn.com/album/my-own-way-hindi-version/M-dvmYgusD4_",
      },
      "artists": {
        "all": [
          {
            "id": "464972",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_500x500.jpg",
              },
            ],
            "name": "Jay Sean",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/jay-sean-songs/8Ea5vNMFq7w_",
          },
        ],
        "featured": [],
        "primary": [
          {
            "id": "464972",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Jay_Sean_500x500.jpg",
              },
            ],
            "name": "Jay Sean",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/jay-sean-songs/8Ea5vNMFq7w_",
          },
        ],
      },
      "copyright": "(C) 2008 2Point9/Jayded",
      "downloadUrl": [
        {
          "quality": "12kbps",
          "url": "https://aac.saavncdn.com/805/4d903cc84643cb2971d9d57b1f146ae8_12.mp4",
        },
        {
          "quality": "48kbps",
          "url": "https://aac.saavncdn.com/805/4d903cc84643cb2971d9d57b1f146ae8_48.mp4",
        },
        {
          "quality": "96kbps",
          "url": "https://aac.saavncdn.com/805/4d903cc84643cb2971d9d57b1f146ae8_96.mp4",
        },
        {
          "quality": "160kbps",
          "url": "https://aac.saavncdn.com/805/4d903cc84643cb2971d9d57b1f146ae8_160.mp4",
        },
        {
          "quality": "320kbps",
          "url": "https://aac.saavncdn.com/805/4d903cc84643cb2971d9d57b1f146ae8_320.mp4",
        },
      ],
      "duration": 193,
      "explicitContent": false,
      "hasLyrics": false,
      "id": "LzlcDECJ",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/805/My-Own-Way-Hindi-Version--English-2008-20191021221531-50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/805/My-Own-Way-Hindi-Version--English-2008-20191021221531-150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/805/My-Own-Way-Hindi-Version--English-2008-20191021221531-500x500.jpg",
        },
      ],
      "label": "TwoPointNine Ltd (Entertainment)",
      "language": "punjabi",
      "lyricsId": null,
      "name": "Ride It (Kya Yehi Pyaar Hai) [Hindi Version]",
      "playCount": Any<Number>,
      "releaseDate": null,
      "type": "song",
      "url": "https://www.jiosaavn.com/song/ride-it-kya-yehi-pyaar-hai-hindi-version/PBIHUjB1dHk",
      "year": 2008,
    },
    {
      "album": {
        "id": "16238551",
        "name": "Ride",
        "url": "https://www.jiosaavn.com/album/ride/1HHayQxPEZI_",
      },
      "artists": {
        "all": [
          {
            "id": "720746",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_500x500.jpg",
              },
            ],
            "name": "Twenty One Pilots",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/twenty-one-pilots-songs/Dx5Vty3hXGs_",
          },
        ],
        "featured": [],
        "primary": [
          {
            "id": "720746",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Twenty_One_Pilots_500x500.jpg",
              },
            ],
            "name": "Twenty One Pilots",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/twenty-one-pilots-songs/Dx5Vty3hXGs_",
          },
        ],
      },
      "copyright": "â„— 2015 Fueled By Ramen LLC",
      "downloadUrl": [
        {
          "quality": "12kbps",
          "url": "https://aac.saavncdn.com/777/b46ecef9fd8d76f4ebadb6a567b029a6_12.mp4",
        },
        {
          "quality": "48kbps",
          "url": "https://aac.saavncdn.com/777/b46ecef9fd8d76f4ebadb6a567b029a6_48.mp4",
        },
        {
          "quality": "96kbps",
          "url": "https://aac.saavncdn.com/777/b46ecef9fd8d76f4ebadb6a567b029a6_96.mp4",
        },
        {
          "quality": "160kbps",
          "url": "https://aac.saavncdn.com/777/b46ecef9fd8d76f4ebadb6a567b029a6_160.mp4",
        },
        {
          "quality": "320kbps",
          "url": "https://aac.saavncdn.com/777/b46ecef9fd8d76f4ebadb6a567b029a6_320.mp4",
        },
      ],
      "duration": 214,
      "explicitContent": false,
      "hasLyrics": false,
      "id": "Hgbp1ser",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/777/Ride-English-2015-20190607044124-50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/777/Ride-English-2015-20190607044124-150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/777/Ride-English-2015-20190607044124-500x500.jpg",
        },
      ],
      "label": "Fueled By Ramen",
      "language": "english",
      "lyricsId": null,
      "name": "Ride",
      "playCount": Any<Number>,
      "releaseDate": null,
      "type": "song",
      "url": "https://www.jiosaavn.com/song/ride/OA8JQUVDUkE",
      "year": 2015,
    },
    {
      "album": {
        "id": "35001223",
        "name": "The Last Ride",
        "url": "https://www.jiosaavn.com/album/the-last-ride/-utCWM9HNsA_",
      },
      "artists": {
        "all": [
          {
            "id": "3319750",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_500x500.jpg",
              },
            ],
            "name": "Sidhu Moose Wala",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/sidhu-moose-wala-songs/ylevcL-ZuH8_",
          },
          {
            "id": "2887201",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_500x500.jpg",
              },
            ],
            "name": "Wazir Patar",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/wazir-patar-songs/t9tPEDdEIGI_",
          },
          {
            "id": "3319750",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_500x500.jpg",
              },
            ],
            "name": "Sidhu Moose Wala",
            "role": "lyricist",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/sidhu-moose-wala-songs/ylevcL-ZuH8_",
          },
        ],
        "featured": [],
        "primary": [
          {
            "id": "3319750",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Sidhu_Moose_Wala_20190627113332_500x500.jpg",
              },
            ],
            "name": "Sidhu Moose Wala",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/sidhu-moose-wala-songs/ylevcL-ZuH8_",
          },
          {
            "id": "2887201",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Wazir_Patar_000_20230809182113_500x500.jpg",
              },
            ],
            "name": "Wazir Patar",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/wazir-patar-songs/t9tPEDdEIGI_",
          },
        ],
      },
      "copyright": "Â© 2022 Sidhu Moose Wala",
      "downloadUrl": [
        {
          "quality": "12kbps",
          "url": "https://aac.saavncdn.com/582/66fbc0a08cf9051e4faad09c541e49ce_12.mp4",
        },
        {
          "quality": "48kbps",
          "url": "https://aac.saavncdn.com/582/66fbc0a08cf9051e4faad09c541e49ce_48.mp4",
        },
        {
          "quality": "96kbps",
          "url": "https://aac.saavncdn.com/582/66fbc0a08cf9051e4faad09c541e49ce_96.mp4",
        },
        {
          "quality": "160kbps",
          "url": "https://aac.saavncdn.com/582/66fbc0a08cf9051e4faad09c541e49ce_160.mp4",
        },
        {
          "quality": "320kbps",
          "url": "https://aac.saavncdn.com/582/66fbc0a08cf9051e4faad09c541e49ce_320.mp4",
        },
      ],
      "duration": 262,
      "explicitContent": false,
      "hasLyrics": true,
      "id": "S-lVIuFT",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/582/The-Last-Ride-Punjabi-2022-20220517061646-50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/582/The-Last-Ride-Punjabi-2022-20220517061646-150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/582/The-Last-Ride-Punjabi-2022-20220517061646-500x500.jpg",
        },
      ],
      "label": "Sidhu Moose Wala",
      "language": "punjabi",
      "lyricsId": null,
      "name": "The Last Ride",
      "playCount": Any<Number>,
      "releaseDate": null,
      "type": "song",
      "url": "https://www.jiosaavn.com/song/the-last-ride/I0UHZz1FcWc",
      "year": 2022,
    },
    {
      "album": {
        "id": "1045211",
        "name": "Ride",
        "url": "https://www.jiosaavn.com/album/ride/danvMf2Tz7U_",
      },
      "artists": {
        "all": [
          {
            "id": "773287",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_500x500.jpg",
              },
            ],
            "name": "Hemachandra",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/hemachandra-songs/sVN5cEMko3I_",
          },
          {
            "id": "456070",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_500x500.jpg",
              },
            ],
            "name": "Sravana Bhargavi",
            "role": "singer",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/sravana-bhargavi-songs/Kv83BqwKExw_",
          },
          {
            "id": "705336",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Aksha_Pardasany_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Aksha_Pardasany_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Aksha_Pardasany_500x500.jpg",
              },
            ],
            "name": "Aksha Pardasany",
            "role": "starring",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/aksha-pardasany-songs/BTdg-FGFqNg_",
          },
          {
            "id": "469841",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/758/O-Pashani-Bengali-2017-50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/758/O-Pashani-Bengali-2017-150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/758/O-Pashani-Bengali-2017-500x500.jpg",
              },
            ],
            "name": "Tanish",
            "role": "starring",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/tanish-songs/PoxH6FTxe,E_",
          },
          {
            "id": "469498",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Nani_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Nani_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Nani_500x500.jpg",
              },
            ],
            "name": "Nani",
            "role": "starring",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/nani-songs/,lgVuTAMD7w_",
          },
        ],
        "featured": [],
        "primary": [
          {
            "id": "773287",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Vedala_Hemachandra_002_20210828063616_500x500.jpg",
              },
            ],
            "name": "Hemachandra",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/hemachandra-songs/sVN5cEMko3I_",
          },
          {
            "id": "456070",
            "image": [
              {
                "quality": "50x50",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_50x50.jpg",
              },
              {
                "quality": "150x150",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_150x150.jpg",
              },
              {
                "quality": "500x500",
                "url": "https://c.saavncdn.com/artists/Sravana_Bhargavi_500x500.jpg",
              },
            ],
            "name": "Sravana Bhargavi",
            "role": "primary_artists",
            "type": "artist",
            "url": "https://www.jiosaavn.com/artist/sravana-bhargavi-songs/Kv83BqwKExw_",
          },
        ],
      },
      "copyright": "Â©  2009 Aditya Music",
      "downloadUrl": [
        {
          "quality": "12kbps",
          "url": "https://aac.saavncdn.com/211/f5c2b6267c69d02170fc55fa1ccb7eb4_12.mp4",
        },
        {
          "quality": "48kbps",
          "url": "https://aac.saavncdn.com/211/f5c2b6267c69d02170fc55fa1ccb7eb4_48.mp4",
        },
        {
          "quality": "96kbps",
          "url": "https://aac.saavncdn.com/211/f5c2b6267c69d02170fc55fa1ccb7eb4_96.mp4",
        },
        {
          "quality": "160kbps",
          "url": "https://aac.saavncdn.com/211/f5c2b6267c69d02170fc55fa1ccb7eb4_160.mp4",
        },
        {
          "quality": "320kbps",
          "url": "https://aac.saavncdn.com/211/f5c2b6267c69d02170fc55fa1ccb7eb4_320.mp4",
        },
      ],
      "duration": 278,
      "explicitContent": false,
      "hasLyrics": false,
      "id": "RNYhaOEc",
      "image": [
        {
          "quality": "50x50",
          "url": "https://c.saavncdn.com/211/Ride-2009-50x50.jpg",
        },
        {
          "quality": "150x150",
          "url": "https://c.saavncdn.com/211/Ride-2009-150x150.jpg",
        },
        {
          "quality": "500x500",
          "url": "https://c.saavncdn.com/211/Ride-2009-500x500.jpg",
        },
      ],
      "label": "Aditya Music",
      "language": "telugu",
      "lyricsId": null,
      "name": "Ride",
      "playCount": Any<Number>,
      "releaseDate": null,
      "type": "song",
      "url": "https://www.jiosaavn.com/song/ride/IiYyWRV-clA",
      "year": 2009,
    },
  ],
  "start": 1,
  "total": Any<Number>,
}
`;
